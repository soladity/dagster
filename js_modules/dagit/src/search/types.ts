export enum SearchResultType {
  Asset,
  Page,
  Pipeline,
  Repository,
  Run,
  Schedule,
  Sensor,
  Solid,
}

export type SearchResult = {
  key: string;
  label: string;
  description: string;
  href: string;
  type: SearchResultType;
  tags?: string;
};
