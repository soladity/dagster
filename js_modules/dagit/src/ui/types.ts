export type Spacing = 0 | 1 | 2 | 4 | 8 | 12 | 16 | 20 | 24 | 32 | 48 | 64;
export type DirectionalSpacing =
  | Spacing
  | {
      top?: Spacing;
      right?: Spacing;
      bottom?: Spacing;
      left?: Spacing;
      vertical?: Spacing;
      horizontal?: Spacing;
    };

export type AlignItems =
  | 'stretch'
  | 'center'
  | 'start'
  | 'end'
  | 'flex-start'
  | 'flex-end'
  | 'baseline';

export type FlexDirection = 'row' | 'row-reverse' | 'column' | 'column-reverse';
export type FlexProperties = {
  alignItems?: AlignItems;
  basis?: string;
  direction?: FlexDirection;
  display?: 'flex' | 'inline-flex';
  grow?: number;
  justifyContent?:
    | 'center'
    | 'start'
    | 'end'
    | 'flex-start'
    | 'flex-end'
    | 'left'
    | 'right'
    | 'space-between'
    | 'space-around'
    | 'space-evenly'
    | 'stretch';
  shrink?: number;
  wrap?: 'nowrap' | 'wrap' | 'wrap-reverse';
};

export type BorderSide = 'top' | 'right' | 'bottom' | 'left' | 'horizontal' | 'vertical' | 'all';
export type BorderWidth = 1 | 2;
export type BorderColor = 'light' | 'medium' | 'dark';
export type BorderSetting = {width: BorderWidth; color: BorderColor; side: BorderSide};
