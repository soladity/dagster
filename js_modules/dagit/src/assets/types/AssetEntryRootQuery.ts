// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AssetKeyInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL query operation: AssetEntryRootQuery
// ====================================================

export interface AssetEntryRootQuery_assetOrError_AssetNotFoundError {
  __typename: "AssetNotFoundError";
}

export interface AssetEntryRootQuery_assetOrError_AssetsNotSupportedError {
  __typename: "AssetsNotSupportedError";
  message: string;
}

export interface AssetEntryRootQuery_assetOrError_Asset_key {
  __typename: "AssetKey";
  path: string[];
}

export interface AssetEntryRootQuery_assetOrError_Asset {
  __typename: "Asset";
  key: AssetEntryRootQuery_assetOrError_Asset_key;
}

export type AssetEntryRootQuery_assetOrError = AssetEntryRootQuery_assetOrError_AssetNotFoundError | AssetEntryRootQuery_assetOrError_AssetsNotSupportedError | AssetEntryRootQuery_assetOrError_Asset;

export interface AssetEntryRootQuery {
  assetOrError: AssetEntryRootQuery_assetOrError;
}

export interface AssetEntryRootQueryVariables {
  assetKey: AssetKeyInput;
}
