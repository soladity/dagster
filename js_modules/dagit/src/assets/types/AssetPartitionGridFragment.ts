// @generated
/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL fragment: AssetPartitionGridFragment
// ====================================================

export interface AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError {
  __typename: "PipelineRunNotFoundError" | "PythonError";
}

export interface AssetPartitionGridFragment_runOrError_PipelineRun {
  __typename: "PipelineRun";
  pipelineSnapshotId: string | null;
}

export type AssetPartitionGridFragment_runOrError = AssetPartitionGridFragment_runOrError_PipelineRunNotFoundError | AssetPartitionGridFragment_runOrError_PipelineRun;

export interface AssetPartitionGridFragment {
  __typename: "AssetMaterialization";
  partition: string | null;
  runOrError: AssetPartitionGridFragment_runOrError;
}
