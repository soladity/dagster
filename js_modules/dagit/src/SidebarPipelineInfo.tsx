import * as React from "react";
import gql from "graphql-tag";
import styled from "styled-components";
import { Colors, Icon } from "@blueprintjs/core";
import {
  SidebarTitle,
  SidebarSubhead,
  SidebarSection,
  SectionItemContainer,
  SectionItemHeader
} from "./SidebarComponents";
import Description from "./Description";
import ConfigTypeSchema from "./ConfigTypeSchema";
import { SidebarPipelineInfoFragment } from "./types/SidebarPipelineInfoFragment";
import { IconNames } from "@blueprintjs/icons";

const NO_DESCRIPTION = "";

interface ISidebarPipelineInfoProps {
  pipeline: SidebarPipelineInfoFragment;
}

export default class SidebarPipelineInfo extends React.Component<
  ISidebarPipelineInfoProps
> {
  static fragments = {
    SidebarPipelineInfoFragment: gql`
      fragment SidebarPipelineInfoFragment on Pipeline {
        name
        description
        contexts {
          name
          description
          config {
            configType {
              ...ConfigTypeSchemaFragment
            }
          }
          resources {
            name
            description
            config {
              configType {
                ...ConfigTypeSchemaFragment
              }
            }
          }
        }
      }

      ${ConfigTypeSchema.fragments.ConfigTypeSchemaFragment}
    `
  };

  render() {
    const { pipeline } = this.props;

    return (
      <div>
        <SidebarSubhead>Pipeline</SidebarSubhead>
        <SidebarTitle>{pipeline.name}</SidebarTitle>
        <SidebarSection title={"Description"}>
          <Description
            description={pipeline ? pipeline.description : NO_DESCRIPTION}
          />
        </SidebarSection>
        <SidebarSection title={"Contexts"}>
          {pipeline.contexts.map(context => (
            <SectionItemContainer key={context.name}>
              <SectionItemHeader>{context.name}</SectionItemHeader>
              <Description
                description={context.description || NO_DESCRIPTION}
              />
              {context.config && (
                <ConfigTypeSchema type={context.config.configType} />
              )}
              {context.resources.map(resource => (
                <ContextResourceContainer key={resource.name}>
                  <Icon
                    iconSize={14}
                    icon={IconNames.LAYERS}
                    color={Colors.DARK_GRAY2}
                  />
                  <div>
                    <ContextResourceHeader>
                      {resource.name}
                    </ContextResourceHeader>
                    <Description
                      description={resource.description || NO_DESCRIPTION}
                    />
                    {resource.config && (
                      <ConfigTypeSchema type={resource.config.configType} />
                    )}
                  </div>
                </ContextResourceContainer>
              ))}
            </SectionItemContainer>
          ))}
        </SidebarSection>
      </div>
    );
  }
}

const ContextResourceHeader = styled(SectionItemHeader)`
  font-size: 13px;
`;

const ContextResourceContainer = styled.div`
  display: flex;
  align-items: flex-start;
  padding-top: 15px;
  & .bp3-icon {
    padding-top: 7px;
    padding-right: 10px;
  }
`;
