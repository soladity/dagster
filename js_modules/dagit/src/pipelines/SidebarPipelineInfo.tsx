import {gql} from '@apollo/client';
import {Colors, Icon} from '@blueprintjs/core';
import {IconNames} from '@blueprintjs/icons';
import * as React from 'react';
import styled from 'styled-components/macro';

import {ConfigTypeSchema, CONFIG_TYPE_SCHEMA_FRAGMENT} from 'src/ConfigTypeSchema';
import {Description} from 'src/Description';
import {breakOnUnderscores} from 'src/app/Util';
import {
  SectionHeader,
  SectionInner,
  SectionItemContainer,
  SidebarSection,
  SidebarSubhead,
  SidebarTitle,
} from 'src/pipelines/SidebarComponents';
import {SidebarPipelineInfoFragment} from 'src/pipelines/types/SidebarPipelineInfoFragment';

const NO_DESCRIPTION = '';

interface ISidebarPipelineInfoProps {
  pipeline: SidebarPipelineInfoFragment;
}

export const SidebarPipelineInfo: React.FC<ISidebarPipelineInfoProps> = ({pipeline}) => (
  <div>
    <SectionInner>
      <SidebarSubhead>Pipeline</SidebarSubhead>
      <SidebarTitle>{breakOnUnderscores(pipeline.name)}</SidebarTitle>
    </SectionInner>
    <SidebarSection title={'Description'}>
      <Description description={pipeline ? pipeline.description : NO_DESCRIPTION} />
    </SidebarSection>
    <SidebarSection title={'Modes'} collapsedByDefault={true}>
      {pipeline.modes.map((mode) => (
        <SectionItemContainer key={mode.name}>
          <SectionHeader>{mode.name}</SectionHeader>
          <Description description={mode.description || NO_DESCRIPTION} />
          {mode.resources.map((resource) => (
            <ContextResourceContainer key={resource.name}>
              <Icon iconSize={14} icon={IconNames.LAYERS} color={Colors.DARK_GRAY2} />
              <div>
                <ContextResourceHeader>{resource.name}</ContextResourceHeader>
                <Description description={resource.description || NO_DESCRIPTION} />
                {resource.configField && (
                  <ConfigTypeSchema
                    type={resource.configField.configType}
                    typesInScope={resource.configField.configType.recursiveConfigTypes}
                  />
                )}
              </div>
            </ContextResourceContainer>
          ))}
          {mode.loggers.map((logger) => (
            <ContextLoggerContainer key={logger.name}>
              <Icon iconSize={14} icon={IconNames.LAYERS} color={Colors.DARK_GRAY2} />
              <div>
                <ContextLoggerHeader>{logger.name}</ContextLoggerHeader>
                <Description description={logger.description || NO_DESCRIPTION} />
                {logger.configField && (
                  <ConfigTypeSchema
                    type={logger.configField.configType}
                    typesInScope={logger.configField.configType.recursiveConfigTypes}
                  />
                )}
              </div>
            </ContextLoggerContainer>
          ))}
        </SectionItemContainer>
      ))}
    </SidebarSection>
  </div>
);

export const SIDEBAR_PIPELINE_INFO_FRAGMENT = gql`
  fragment SidebarPipelineInfoFragment on IPipelineSnapshot {
    name
    description
    modes {
      name
      description
      resources {
        name
        description
        configField {
          configType {
            ...ConfigTypeSchemaFragment
            recursiveConfigTypes {
              ...ConfigTypeSchemaFragment
            }
          }
        }
      }
      loggers {
        name
        description
        configField {
          configType {
            ...ConfigTypeSchemaFragment
            recursiveConfigTypes {
              ...ConfigTypeSchemaFragment
            }
          }
        }
      }
    }
  }

  ${CONFIG_TYPE_SCHEMA_FRAGMENT}
`;

const ContextResourceHeader = styled(SectionHeader)`
  font-size: 13px;
`;

const ContextResourceContainer = styled.div`
  display: flex;
  align-items: flex-start;
  padding-top: 15px;
  & .bp3-icon {
    padding-top: 7px;
    padding-right: 10px;
  }
`;

const ContextLoggerHeader = styled(SectionHeader)`
  font-size: 13px;
`;

const ContextLoggerContainer = styled.div`
  display: flex;
  align-items: flex-start;
  padding-top: 15px;
  & .bp3-icon {
    padding-top: 7px;
    padding-right: 10px;
  }
`;
