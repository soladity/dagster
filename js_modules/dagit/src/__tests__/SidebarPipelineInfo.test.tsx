import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import SidebarPipelineInfo from "../SidebarPipelineInfo";
import { SidebarPipelineInfoFragment } from "../types/SidebarPipelineInfoFragment";

it("renders given a pipeline with resources and loggers", () => {
  const sidebarPipelineInfoData: SidebarPipelineInfoFragment = {
    __typename: "Pipeline",
    name: "A good test pipeline",
    description: "Very friendly and helpful",
    modes: [
      {
        __typename: "Mode",
        name: "one_good_mode",
        description: "A way to run your pipeline",
        resources: [
          {
            __typename: "Resource",
            name: "a_resource",
            description: "Very useful",
            configField: {
              __typename: "ConfigTypeField",
              configType: {
                __typename: "NullableConfigType",
                name: "config",
                key: "Dict.88",
                description: "A configuration dictionary with typed fields",
                innerTypes: [
                  {
                    __typename: "RegularConfigType",
                    name: "foo",
                    key: "Int",
                    description: null,
                    innerTypes: [],
                    isNullable: false,
                    isList: false,
                    isSelector: false
                  }
                ],
                isList: false,
                isNullable: true,
                isSelector: false
              }
            }
          }
        ],
        loggers: []
      },
      {
        __typename: "Mode",
        name: "another_mode",
        description: "Another nice way to run your pipeline",
        resources: [
          {
            __typename: "Resource",
            name: "a_resource",
            description: "Very useful",
            configField: {
              __typename: "ConfigTypeField",
              configType: {
                __typename: "NullableConfigType",
                name: "config",
                key: "Dict.88",
                description: "A configuration dictionary with typed fields",
                innerTypes: [
                  {
                    __typename: "RegularConfigType",
                    name: "foo",
                    key: "Int",
                    description: null,
                    innerTypes: [],
                    isNullable: false,
                    isList: false,
                    isSelector: false
                  }
                ],
                isList: false,
                isNullable: true,
                isSelector: false
              }
            }
          },
          {
            __typename: "Resource",
            name: "b_resource",
            description: "Very useful",
            configField: {
              __typename: "ConfigTypeField",
              configType: {
                __typename: "NullableConfigType",
                name: "config",
                key: "Dict.88",
                description: "A configuration dictionary with typed fields",
                innerTypes: [
                  {
                    __typename: "RegularConfigType",
                    name: "foo",
                    key: "Int",
                    description: null,
                    innerTypes: [],
                    isNullable: false,
                    isList: false,
                    isSelector: false
                  }
                ],
                isList: false,
                isNullable: true,
                isSelector: false
              }
            }
          }
        ],
        loggers: [
          {
            __typename: "Logger",
            name: "console",
            description: "The default colored console logger.",
            configField: {
              __typename: "ConfigTypeField",
              configType: {
                __typename: "CompositeConfigType",
                key: "Dict.23",
                name: null,
                description: "A configuration dictionary with typed fields",
                fields: [
                  {
                    __typename: "ConfigTypeField",
                    name: "log_level",
                    description: null,
                    configType: {
                      __typename: "RegularConfigType",
                      key: "String"
                    },
                    isOptional: true
                  },
                  {
                    __typename: "ConfigTypeField",
                    name: "name",
                    description: null,
                    configType: {
                      __typename: "RegularConfigType",
                      key: "String"
                    },
                    isOptional: true
                  }
                ],
                innerTypes: [
                  {
                    __typename: "RegularConfigType",
                    name: "String",
                    key: "String",
                    description: "",
                    isList: false,
                    isNullable: false,
                    isSelector: false,
                    innerTypes: []
                  }
                ],
                isNullable: false,
                isList: false,
                isSelector: false
              }
            }
          }
        ]
      }
    ]
  };
  const component = TestRenderer.create(
    <BrowserRouter>
      <SidebarPipelineInfo pipeline={sidebarPipelineInfoData} />
    </BrowserRouter>
  );
  expect(component.toJSON()).toMatchSnapshot();
});
