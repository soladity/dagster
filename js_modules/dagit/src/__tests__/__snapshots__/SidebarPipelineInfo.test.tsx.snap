// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders given a pipeline with resources and loggers 1`] = `
<div>
  <div
    className="sc-htoDjs icLeVm"
  >
    <div
      className="sc-EHOje dltgwV"
    >
      Pipeline
    </div>
    <h3
      className="sc-bwzfXH kAXACy"
    >
      A good test pipeline
    </h3>
  </div>
  <div>
    <div
      className="sc-gzVnrw kmnULz"
      onClick={[Function]}
    >
      Description
      <span
        className="bp3-icon bp3-icon-chevron-down sc-bdVaJa jvnmTD"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      className="bp3-collapse"
      style={
        Object {
          "height": "auto",
          "overflowY": "visible",
          "transition": "none",
        }
      }
    >
      <div
        aria-hidden={false}
        className="bp3-collapse-body"
        style={
          Object {
            "transform": "translateY(0)",
            "transition": "none",
          }
        }
      >
        <div
          className="sc-htoDjs icLeVm"
        >
          <div
            className="sc-dnqmqq jAGQAf"
            onDoubleClick={[Function]}
            style={
              Object {
                "maxHeight": 320,
              }
            }
          >
            <div
              style={
                Object {
                  "overflowX": "auto",
                }
              }
            >
              <p>
                Very friendly and helpful
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      className="sc-gzVnrw kmnULz"
      onClick={[Function]}
    >
      Modes
      <span
        className="bp3-icon bp3-icon-chevron-down sc-bdVaJa jvnmTD"
        icon="chevron-down"
      >
        <svg
          data-icon="chevron-down"
          height={16}
          viewBox="0 0 16 16"
          width={16}
        >
          <desc>
            chevron-down
          </desc>
          <path
            d="M12 5c-.28 0-.53.11-.71.29L8 8.59l-3.29-3.3a1.003 1.003 0 00-1.42 1.42l4 4c.18.18.43.29.71.29s.53-.11.71-.29l4-4A1.003 1.003 0 0012 5z"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </div>
    <div
      className="bp3-collapse"
      style={
        Object {
          "height": "auto",
          "overflowY": "visible",
          "transition": "none",
        }
      }
    >
      <div
        aria-hidden={false}
        className="bp3-collapse-body"
        style={
          Object {
            "transform": "translateY(0)",
            "transition": "none",
          }
        }
      >
        <div
          className="sc-htoDjs icLeVm"
        >
          <div
            className="sc-bZQynM dPnpZk"
          >
            <h4
              className="sc-htpNat gNWcHK"
            >
              one_good_mode
            </h4>
            <div
              className="sc-dnqmqq jAGQAf"
              onDoubleClick={[Function]}
              style={
                Object {
                  "maxHeight": 320,
                }
              }
            >
              <div
                style={
                  Object {
                    "overflowX": "auto",
                  }
                }
              >
                <p>
                  A way to run your pipeline
                </p>
              </div>
            </div>
            <div
              className="sc-cSHVUG lgPeWV"
            >
              <span
                className="bp3-icon bp3-icon-layers"
                icon="layers"
              >
                <svg
                  data-icon="layers"
                  fill="#202B33"
                  height={14}
                  viewBox="0 0 16 16"
                  width={14}
                >
                  <desc>
                    layers
                  </desc>
                  <path
                    d="M.55 4.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5a.998.998 0 00-.06-1.81L8.4.08a1.006 1.006 0 00-.79 0l-6.99 3a.992.992 0 00-.07 1.81zM15 10c-.16 0-.31.04-.45.11L8 13.38 1.45 10.1c-.14-.06-.29-.1-.45-.1-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1zm0-3.5c-.16 0-.31.04-.45.11L8 9.88 1.45 6.61A.997.997 0 001 6.5c-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <div>
                <h4
                  className="sc-htpNat sc-jzJRlG jNaeoS"
                >
                  a_resource
                </h4>
                <div
                  className="sc-dnqmqq jAGQAf"
                  onDoubleClick={[Function]}
                  style={
                    Object {
                      "maxHeight": 320,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "overflowX": "auto",
                      }
                    }
                  >
                    <p>
                      Very useful
                    </p>
                  </div>
                </div>
                <code
                  className="sc-gqjmRU bRLIya"
                >
                  <div
                    className="sc-fjdhpX edikPP"
                  >
                    /* A configuration dictionary with typed fields */
                  </div>
                  <span>
                    foo
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#BF7326",
                        "fontWeight": 500,
                      }
                    }
                  >
                    ?
                  </span>
                </code>
              </div>
            </div>
          </div>
          <div
            className="sc-bZQynM dPnpZk"
          >
            <h4
              className="sc-htpNat gNWcHK"
            >
              another_mode
            </h4>
            <div
              className="sc-dnqmqq jAGQAf"
              onDoubleClick={[Function]}
              style={
                Object {
                  "maxHeight": 320,
                }
              }
            >
              <div
                style={
                  Object {
                    "overflowX": "auto",
                  }
                }
              >
                <p>
                  Another nice way to run your pipeline
                </p>
              </div>
            </div>
            <div
              className="sc-cSHVUG lgPeWV"
            >
              <span
                className="bp3-icon bp3-icon-layers"
                icon="layers"
              >
                <svg
                  data-icon="layers"
                  fill="#202B33"
                  height={14}
                  viewBox="0 0 16 16"
                  width={14}
                >
                  <desc>
                    layers
                  </desc>
                  <path
                    d="M.55 4.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5a.998.998 0 00-.06-1.81L8.4.08a1.006 1.006 0 00-.79 0l-6.99 3a.992.992 0 00-.07 1.81zM15 10c-.16 0-.31.04-.45.11L8 13.38 1.45 10.1c-.14-.06-.29-.1-.45-.1-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1zm0-3.5c-.16 0-.31.04-.45.11L8 9.88 1.45 6.61A.997.997 0 001 6.5c-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <div>
                <h4
                  className="sc-htpNat sc-jzJRlG jNaeoS"
                >
                  a_resource
                </h4>
                <div
                  className="sc-dnqmqq jAGQAf"
                  onDoubleClick={[Function]}
                  style={
                    Object {
                      "maxHeight": 320,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "overflowX": "auto",
                      }
                    }
                  >
                    <p>
                      Very useful
                    </p>
                  </div>
                </div>
                <code
                  className="sc-gqjmRU bRLIya"
                >
                  <div
                    className="sc-fjdhpX edikPP"
                  >
                    /* A configuration dictionary with typed fields */
                  </div>
                  <span>
                    foo
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#BF7326",
                        "fontWeight": 500,
                      }
                    }
                  >
                    ?
                  </span>
                </code>
              </div>
            </div>
            <div
              className="sc-cSHVUG lgPeWV"
            >
              <span
                className="bp3-icon bp3-icon-layers"
                icon="layers"
              >
                <svg
                  data-icon="layers"
                  fill="#202B33"
                  height={14}
                  viewBox="0 0 16 16"
                  width={14}
                >
                  <desc>
                    layers
                  </desc>
                  <path
                    d="M.55 4.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5a.998.998 0 00-.06-1.81L8.4.08a1.006 1.006 0 00-.79 0l-6.99 3a.992.992 0 00-.07 1.81zM15 10c-.16 0-.31.04-.45.11L8 13.38 1.45 10.1c-.14-.06-.29-.1-.45-.1-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1zm0-3.5c-.16 0-.31.04-.45.11L8 9.88 1.45 6.61A.997.997 0 001 6.5c-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <div>
                <h4
                  className="sc-htpNat sc-jzJRlG jNaeoS"
                >
                  b_resource
                </h4>
                <div
                  className="sc-dnqmqq jAGQAf"
                  onDoubleClick={[Function]}
                  style={
                    Object {
                      "maxHeight": 320,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "overflowX": "auto",
                      }
                    }
                  >
                    <p>
                      Very useful
                    </p>
                  </div>
                </div>
                <code
                  className="sc-gqjmRU bRLIya"
                >
                  <div
                    className="sc-fjdhpX edikPP"
                  >
                    /* A configuration dictionary with typed fields */
                  </div>
                  <span>
                    foo
                  </span>
                  <span
                    style={
                      Object {
                        "color": "#BF7326",
                        "fontWeight": 500,
                      }
                    }
                  >
                    ?
                  </span>
                </code>
              </div>
            </div>
            <div
              className="sc-chPdSV kBkKzM"
            >
              <span
                className="bp3-icon bp3-icon-layers"
                icon="layers"
              >
                <svg
                  data-icon="layers"
                  fill="#202B33"
                  height={14}
                  viewBox="0 0 16 16"
                  width={14}
                >
                  <desc>
                    layers
                  </desc>
                  <path
                    d="M.55 4.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5a.998.998 0 00-.06-1.81L8.4.08a1.006 1.006 0 00-.79 0l-6.99 3a.992.992 0 00-.07 1.81zM15 10c-.16 0-.31.04-.45.11L8 13.38 1.45 10.1c-.14-.06-.29-.1-.45-.1-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1zm0-3.5c-.16 0-.31.04-.45.11L8 9.88 1.45 6.61A.997.997 0 001 6.5c-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1z"
                    fillRule="evenodd"
                  />
                </svg>
              </span>
              <div>
                <h4
                  className="sc-htpNat sc-kAzzGY frhvrD"
                >
                  console
                </h4>
                <div
                  className="sc-dnqmqq jAGQAf"
                  onDoubleClick={[Function]}
                  style={
                    Object {
                      "maxHeight": 320,
                    }
                  }
                >
                  <div
                    style={
                      Object {
                        "overflowX": "auto",
                      }
                    }
                  >
                    <p>
                      The default colored console logger.
                    </p>
                  </div>
                </div>
                <code
                  className="sc-gqjmRU bRLIya"
                >
                  <div
                    className="sc-fjdhpX edikPP"
                  >
                    /* A configuration dictionary with typed fields */
                  </div>
                  {
                  <div
                    className="sc-VigVT fIhGog"
                  >
                      
                    <span
                      className="sc-jTzLTM eSIHQs"
                    >
                      log_level
                    </span>
                    <span
                      style={
                        Object {
                          "color": "#BF7326",
                          "fontWeight": 500,
                        }
                      }
                    >
                      ?
                    </span>
                    : 
                    <span>
                      String
                    </span>
                  </div>
                  <div
                    className="sc-VigVT fIhGog"
                  >
                      
                    <span
                      className="sc-jTzLTM eSIHQs"
                    >
                      name
                    </span>
                    <span
                      style={
                        Object {
                          "color": "#BF7326",
                          "fontWeight": 500,
                        }
                      }
                    >
                      ?
                    </span>
                    : 
                    <span>
                      String
                    </span>
                  </div>
                  }
                </code>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
